Velaptor/Batching/BatchingManager.cs

```diff
@@ -0,0 +1,319 @@
+            respondId: PullResponses.GetFontItemsId,
+            name: fontPullReactorName,
+            onRespond: () =>
+            {
+                var lastFullItemIndex = this.fontItems.IndexOf(i => i.IsEmpty());
+
+                return lastFullItemIndex < 0
+                    ? default
+                    : this.fontItems[..lastFullItemIndex];
+            }));
+
+        var rectPullReactable = reactableFactory.CreateRectPullBatchReactable();
+        var rectPullReactorName = this.GetExecutionMemberName(nameof(PullResponses.GetRectItemsId));
+        this.rectPullUnsubscriber = rectPullReactable.Subscribe(new RespondReactor<Memory<RenderItem<RectBatchItem>>>(
+            respondId: PullResponses.GetRectItemsId,
+            name: rectPullReactorName,
+            onRespond: () =>
+            {
+                var lastFullItemIndex = this.rectItems.IndexOf(i => i.IsEmpty());
+
+                return lastFullItemIndex < 0
+                    ? default
+                    : this.rectItems[..lastFullItemIndex];
+            }));
+
+        var linePullReactable = reactableFactory.CreateLinePullBatchReactable();
+        var linePullReactorName = this.GetExecutionMemberName(nameof(PullResponses.GetLineItemsId));
+        this.linePullUnsubscriber = linePullReactable.Subscribe(new RespondReactor<Memory<RenderItem<LineBatchItem>>>(
+            respondId: PullResponses.GetLineItemsId,
+            name: linePullReactorName,
+            onRespond: () =>
+            {
+                var lastFullItemIndex = this.lineItems.IndexOf(i => i.IsEmpty());
+
+                return lastFullItemIndex < 0
+                    ? default
+                    : this.lineItems[..lastFullItemIndex];
+            }));
+    }
+
+    /// <summary>
+    /// Gets the texture items.
+    /// </summary>
+    /// <remarks>USED FOR UNIT TESTING.</remarks>
+    public Span<RenderItem<TextureBatchItem>> TextureItems => this.textureItems.Span;
+
+    /// <summary>
+    /// Gets the font items.
+    /// </summary>
+    /// <remarks>USED FOR UNIT TESTING.</remarks>
+    public Span<RenderItem<FontGlyphBatchItem>> FontItems => this.fontItems.Span;
+
+    /// <summary>
+    /// Gets the rectangle items.
+    /// </summary>
+    /// <remarks>USED FOR UNIT TESTING.</remarks>
+    public Span<RenderItem<RectBatchItem>> RectItems => this.rectItems.Span;
+
+    /// <summary>
+    /// Gets the line items.
+    /// </summary>
+    /// <remarks>USED FOR UNIT TESTING.</remarks>
+    public Span<RenderItem<LineBatchItem>> LineItems => this.lineItems.Span;
+
+    /// <inheritdoc/>
+    public void AddTextureItem(TextureBatchItem item, int layer)
+    {
+        var emptyItemIndex = this.textureItems.
+            FirstItemIndex(i => i.Item.IsEmpty());
+
+        if (emptyItemIndex == -1)
+        {
+            // TODO: Replace this code and comment once GPU adjustment feature is added
```

Comment:
Is this TODO still needed?  Check for all TODO's...there are more. 
```suggestion
            // TODO: Replace this code and comment once GPU adjustment feature is added
```